src/rules/require-hook.ts

```diff
@@ -48,7 +60,10 @@ const shouldBeInHook = (node: TSESTree.Node): boolean => {
   }
 };
 
-export default createRule({
+export default createRule<
+  [Partial<{ allowedFunctionCalls?: readonly string[] }>],
```

Comment:
```suggestion
  [{ allowedFunctionCalls?: readonly string[] }],
```

nit: `Partial` marks all properties as optional, so we don't need it if the properties already are optional and likewise we don't need to mark the properties as optional if we're using `Partial` 

If we had more properties, using `Partial` could be nicer but for now since we've just got the one we can make the code a little smaller by marking the sole property as optional.